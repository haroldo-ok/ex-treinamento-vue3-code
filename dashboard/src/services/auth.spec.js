import mockAxios from 'axios'
import { Promise } from 'core-js'
import AuthService from './auth'

jest.mock('axios')

describe('AuthService', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should return a token when user logs in', async () => {
    const token = '123.123.123'
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: { token } })
    })

    const response = await AuthService(mockAxios).login({
      email: 'test@example.com',
      password: '1234'
    })

    expect(response.data).toHaveProperty('token')
    expect(response).toMatchSnapshot()
  })

  it('should return an user when user registers', async () => {
    const user = {
      name: 'Bob',
      password: '123',
      email: 'teste@example.com'
    }
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: user })
    })

    const response = await AuthService(mockAxios).register(user)

    expect(response.data).toHaveProperty('name')
    expect(response.data).toHaveProperty('password')
    expect(response.data).toHaveProperty('email')
    expect(response).toMatchSnapshot()
  })

  it('should return an error when not found', async () => {
    const errors = { status: 404, statusText: 'Not found' }
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ request: errors })
    })

    const response = await AuthService(mockAxios).login({
      email: 'test@example.com',
      password: '1234'
    })

    expect(response.errors).toHaveProperty('status')
    expect(response.errors).toHaveProperty('statusText')
  })
})
