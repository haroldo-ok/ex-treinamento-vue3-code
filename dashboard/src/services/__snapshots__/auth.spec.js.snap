// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthService should return a token when user logs in 1`] = `
Object {
  "data": Object {
    "token": "123.123.123",
  },
  "errors": null,
}
`;

exports[`AuthService should return an error when not found 1`] = `
Object {
  "data": undefined,
  "errors": Object {
    "status": 404,
    "statusText": "Not found",
  },
}
`;

exports[`AuthService should return an user when user registers 1`] = `
Object {
  "data": Object {
    "email": "teste@example.com",
    "name": "Bob",
    "password": "123",
  },
  "errors": null,
}
`;
